sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuApproximatePercentile.scala

```diff
@@ -0,0 +1,220 @@
+
+  // initialValues is only used in reduction and this is not currently supported
+  override lazy val initialValues: Seq[GpuExpression] = throw new UnsupportedOperationException(
+    "approx_percentile does not support reduction")
+
+  // the update expression will create a t-digest (List[Struct[Double, Double])
+  override lazy val updateExpressions: Seq[Expression] =
+    new CudfTDigest(inputBuf,
+      percentageExpression,
+      accuracyExpression) :: Nil
+
+  // the merge expression will merge t-digests
+  override lazy val mergeExpressions: Seq[Expression] =
+    new CudfTDigest(outputBuf,
+      percentageExpression,
+      accuracyExpression) :: Nil
+
+  // the evaluate expression will compute percentiles based on a t-digest
+  override lazy val evaluateExpression: Expression = {
+    ApproxPercentileFromTDigestExpr(outputBuf, percentiles, child.dataType)
+  }
+
+  // inputBuf represents the initial t-digest aggregation buffer
+  protected final lazy val inputBuf: AttributeReference =
+    AttributeReference("inputBuf", CudfTDigest.dataType)()
+
+  // outputBuf represents the merged t-digest aggregation buffer
+  protected final lazy val outputBuf: AttributeReference =
+    inputBuf.copy("outputBuf", CudfTDigest.dataType)(inputBuf.exprId, inputBuf.qualifier)
+
+  override def nullable: Boolean = false
+
+  // Mark as lazy so that percentageExpression is not evaluated during tree transformation.
+  private lazy val (returnPercentileArray, percentiles) = makeArray(percentageExpression.value)
+
+  private def makeArray(v: Any): (Boolean, Either[Double, Array[Double]])  = v match {
+    // Rule ImplicitTypeCasts can cast other numeric types to double
+    case null => (false, Right(Array()))
+    case num: Double => (false, Left(num))
+    case arrayData: ArrayData => (true, Right(arrayData.toDoubleArray()))
+    case other => throw new IllegalStateException(s"Invalid percentile expression $other")
+  }
+
+  // The result type is the same as the input type.
+  private lazy val internalDataType: DataType = {
+    if (returnPercentileArray) {
+      ArrayType(child.dataType, containsNull = false)
+    } else {
+      child.dataType
+    }
+  }
+
+  override def dataType: DataType = internalDataType
+
+  override def children: Seq[Expression] = Seq(child, percentageExpression, accuracyExpression)
+}
+
+/**
+ * This expression computes an approximate percentile using a t-digest as input.
+ *
+ * @param child Expression that produces the t-digests.
+ * @param percentiles Percentile scalar, or percentiles array to evaluate.
+ * @param finalDataType Data type for results
+ */
+case class ApproxPercentileFromTDigestExpr(
+    child: Expression,
+    percentiles: Either[Double, Array[Double]],
+    finalDataType: DataType)
+  extends GpuExpression {
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    val expr = child.asInstanceOf[GpuExpression]
+    withResource(expr.columnarEval(batch).asInstanceOf[GpuColumnVector]) { cv =>
```

Comment:
I don't think this is guaranteed to be a `GpuColumnVector`. Can we use `GpuExpressionsUtils.columnarEvalToColumn`